tests/integration/sharding_tests/test_sharding_backups.py

```diff
@@ -232,13 +255,28 @@ async def test_rotate_backup_password(ops_test: OpsTest) -> None:
 
 @pytest.mark.group(1)
 @pytest.mark.abort_on_fail
-async def test_restore_backup(ops_test: OpsTest, add_writes_to_db) -> None:
+async def test_restore_backup(ops_test: OpsTest, add_writes_to_shards) -> None:
     """Tests that sharded Charmed MongoDB cluster supports restores."""
     # count total writes
     cluster_writes = await writes_helpers.get_cluster_writes_count(
         ops_test, shard_app_names=SHARD_APPS
     )
-    assert cluster_writes["total_writes"] > 0, "no writes to backup"
+    shard_one_writes = await writes_helpers.count_shard_writes(
+        ops_test,
+        shard_app_name=SHARD_ONE_APP_NAME,
+        db_name=SHARD_ONE_DB_NAME,
+        collection_name=SHARD_ONE_COLL_NAME,
+    )
+    shard_two_writes = await writes_helpers.count_shard_writes(
+        ops_test,
+        shard_app_name=SHARD_TWO_APP_NAME,
+        db_name=SHARD_TWO_DB_NAME,
+        collection_name=SHARD_TWO_COLL_NAME,
+    )
+
+    assert cluster_writes["total_writes"], "no writes to backup"
+    assert shard_one_writes, "no writes to backup for shard one"
+    assert shard_two_writes, "no writes to backup for shard two"
 
```

Comment:
Can we add something along the lines of: 
```suggestion
   assert shard_one_writes + shard_two_writes >= cluster_writes["total_writes"], "writes not synced"
```

Reply:
yes! changed it to `==` but that was because I changed the count function